pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -353,7 +353,17 @@ cdef class _Timestamp(ABCTimestamp):
             raise NotImplementedError(self._reso)
 
         if is_any_td_scalar(other):
-            nanos = delta_to_nanoseconds(other)
+            if (
+                is_timedelta64_object(other)
+                and get_datetime64_unit(other) == NPY_DATETIMEUNIT.NPY_FR_GENERIC
+            ):
+                # TODO: deprecate allowing this?  We only get here
```

Comment:
Deprecate specifically adding a `np.timedelta64` object without a unit (and not the interval adding behavior)?

If so, +1.

Reply:
correct